ckanext/validation/tests/test_form.py

```diff
@@ -386,245 +275,193 @@ def test_resource_form_create(self, app):
+        If schema and format is provided - resource will be validated according
+        to the schema"""
         dataset = Dataset()
 
-        upload = ('upload', 'valid.csv', VALID_CSV)
-
-        valid_stream = io.BufferedReader(io.BytesIO(VALID_CSV))
-
-        params = {
+        data = {
             'name': 'test_resource_form_create_valid',
-            'url': 'https://example.com/data.csv'
+            'url': 'https://example.com/data.csv',
+            'schema': json.dumps(SCHEMA),
+            'format': 'csv'
         }
 
-        with mock.patch('io.open', return_value=valid_stream):
-            _post(self.app, NEW_RESOURCE_URL.format(dataset['id']),
-                  params, upload=[upload])
+        _post(app,
+              NEW_RESOURCE_URL.format(dataset['id']),
+              data,
+              upload=[('upload', 'data.csv', VALID_CSV)])
 
         dataset = call_action('package_show', id=dataset['id'])
 
         assert dataset['resources'][0]['validation_status'] == 'success'
         assert 'validation_timestamp' in dataset['resources'][0]
 
-    @unittest.skip("TODO Fix file mocks so this can run")
-    @pytest.mark.ckan_config('ckanext.validation.run_on_create_sync', True)
-    @pytest.mark.ckan_config('ckanext.validation.run_on_update_sync', True)
-    @mock_uploads
-    def test_resource_form_create_invalid(self, mock_open):
-        user = Sysadmin()
-        org = Organization(user=user)
-        dataset = Dataset(owner_org=org['id'])
-
-        upload = ('upload', 'invalid.csv', INVALID_CSV)
-
-        invalid_stream = io.BufferedReader(io.BytesIO(INVALID_CSV))
+    def test_resource_form_create_invalid(self, app):
+        """Test we aren't able to create resource with an ivalid CSV file.
+        If schema and format is provided - resource will be validated according
+        to the schema"""
+        dataset = Dataset()
 
-        params = {
+        data = {
             'name': 'test_resource_form_create_invalid',
-            'url': 'https://example.com/data.csv'
+            'url': 'https://example.com/data.csv',
+            'schema': json.dumps(SCHEMA),
+            'format': 'csv'
         }
 
-        with mock.patch('io.open', return_value=invalid_stream):
-            response = _get_response_body(_post(self.app, NEW_RESOURCE_URL.format(dataset['id']),
-                                          params, upload=[upload]))
+        response = _get_response_body(
+            _post(app,
+                  NEW_RESOURCE_URL.format(dataset['id']),
+                  data,
+                  upload=[('upload', 'data.csv', INVALID_CSV)]))
 
         assert 'validation' in response
         assert 'missing-value' in response
         assert 'Row 2 has a missing value in column 4' in response
 
 
 @pytest.mark.usefixtures("clean_db", "validation_setup")
-class TestResourceValidationOnUpdateForm(FunctionalTestBase):
+class TestResourceValidationOnUpdateForm(object):
```

Comment:
why swap out FunctionalTestBase for object?

Reply:
FunctionalTestBase is deprecated: https://github.com/ckan/ckan/blob/5840e49052dc6c747316fedcc7821431f2ec50a0/ckan/tests/helpers.py#L308

Reply:
No sense to use it any more.